docs/tutorials/data_manipulation/dataframesdataset.md

```diff
@@ -0,0 +1,375 @@
+    evaluator = Evaluator(quantiles=(np.arange(20) / 20.0)[1:])
+    agg_metrics, item_metrics = evaluator(ts_it, forecast_it, num_series=len(ds))
+    return agg_metrics["MSE"]
+```
+
+Lets now generate single time series and multiple time series
+(as list and as long-dataframe).
+Note we don't provide `freq` and `timestamp`, because they are automatically
+inferred from the index. We are also not passing "target" and "item_id"
+(for long-dataframe) because those are the default values of `target`
+and `item_id`, respectively.
+
+
+```python
+single_ts = generate_single_ts_with_features(date_range, item_id=0)
+multiple_ts = [generate_single_ts_with_features(date_range, item_id=i) for i in range(N)]
+multiple_ts_long = pd.concat(multiple_ts)
+
+single_ts_dataset = DataFramesDataset(
+    single_ts,
+    feat_dynamic_real=["dynamic_real_1", "dynamic_real_2"],
+    feat_static_cat=["static_cat_1"],
+)
+multiple_ts_dataset = DataFramesDataset(
+    multiple_ts,
+    feat_dynamic_real=["dynamic_real_1", "dynamic_real_2"],
+    feat_static_cat=["static_cat_1"],
+)
+multiple_ts_dataset_dict = DataFramesDataset(
+    {i: _ts for i, _ts in enumerate(multiple_ts)},
+    feat_dynamic_real=["dynamic_real_1", "dynamic_real_2"],
+    feat_static_cat=["static_cat_1"],
+)
+# for long-dataset we use a different constructor and need a `item_id` column
+multiple_ts_long_dataset = DataFramesDataset.from_long_dataframe(
+    multiple_ts_long,
+    feat_dynamic_real=["dynamic_real_1", "dynamic_real_2"],
+    feat_static_cat=["static_cat_1"],
+)
+```
+
+Thats it! We can now call `train_and_predict_with_features` with all the datasets.
+
+
+```python
+train_and_predict_with_features(single_ts_dataset, prediction_length)
+```
+
+```python
+train_and_predict_with_features(multiple_ts_dataset, prediction_length)
+```
+
+```python
+train_and_predict_with_features(multiple_ts_dataset_dict, prediction_length)
+```
+
+```python
+train_and_predict_with_features(multiple_ts_long_dataset, prediction_length)
+```
+
+## Use train/test split
+
+Here, we split the DataFrame/DataFrames into training and test data
+using the `OffsetSplitter` from gluonts. We can then
+use the training data to train the model and the test data for prediction.
+Note, the `OffsetSplitter` needs the `item_id`. If you want to split
+by date you can also use the `DateSplitter`.
+
+
+```python
+prediction_length, freq = 24, "1H"
+T = 10*prediction_length
+date_range = pd.date_range("2021-01-01", periods=T, freq=freq)
+ts = generate_single_ts(date_range, item_id=0)
+```
+
+
+```python
+from gluonts.dataset.split import OffsetSplitter # DateSplitter
+
+#Note the minus-sign for `split_offset`. This will split the time series counting from the end.
+splitter = OffsetSplitter(prediction_length=prediction_length, split_offset=-prediction_length)
+split = splitter.split(DataFramesDataset(ts, item_id="item_id"))
+train, test = split.train, split.test
+```
```

Comment:
This is once again generating a certain dataframe, and converting it to a dataset using `DataFramesDataset`: couldn't this use one of the previously generated dataframes or dataset? So that the snippets here can focus on what `OffsetSplitter` does